task/attest/src/main.rs

```diff
@@ -0,0 +1,185 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Root of trust for reporting (RoT-R) task.
+//!
+//! Use the rotr-api crate to interact with this task.
+
+#![no_std]
+#![no_main]
+
+use attest_api::AttestError;
+use core::mem::MaybeUninit;
+use dice::{AliasData, CertData, FwidCert, FWID_LENGTH};
+use idol_runtime::{ClientError, Leased, RequestError, W};
+use ringbuf::{ringbuf, ringbuf_entry};
+use stage0_handoff::HandoffData;
+use userlib::UnwrapLite;
+use zerocopy::AsBytes;
+
+#[derive(Copy, Clone, PartialEq)]
+enum Trace {
+    Cert,
+    CertChainLen(u32),
+    CertLen(usize),
+    Error(AttestError),
+    Fwid([u8; FWID_LENGTH]),
+    BufSize(usize),
+    Index(u32),
+    Offset(usize),
+    Startup,
+    None,
+}
+
+ringbuf!(Trace, 16, Trace::None);
+
+// Map the memory used to pass the segment of the identity cert chain common
+// to all tasks to a variable.
+#[used]
```

Comment:
You're doing a somewhat subtle thing here that I think should get added to the comment (because I had to cursor around and convince myself it was correct).

By assigning a `link_section` you're saying you want this placed in an area of memory -- _not_ at the base of that memory. However, the app config (and thus the linker script) sizes these two sections of memory such that they're exactly the size of these statics. This should ensure that they're placed at the base of the section.

So, I'd mention that intent in the comment, lest someone adjust it later and cause this to become invalid.

This would probably be better served by the `extern-regions` facility, which helps to avoid this class of thing.

Reply:
Spent some time reading up on the `extern-regions` stuff over the last few days. This PR has been updated to replace the existing code using `link_section`s with `extern-regions`. This took a bit more stack space but there's probably ways to reduce that if necessary (mut statics?).